### PR TITLE
ODC-7498: Show an opt-in / opt-out option to the user depending on the cluster configuration

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Story**: https://issues.redhat.com/browse/ODC-7498

**Description**:

Added `Telemetry`  tab in cluster configuration page, which have 4 options, `Opt-in, Opt-out, Enforce and Disabled`

Screenshot 2024-02-07 at 11 05 19 AM

-------

Added `Telemetry` Section under `User Preferences--> General tab` where the user can select `Accept or Deny` sending telemetry events and this section will be **enabled only if** the cluster telemetry setting is set to **Opt-in or Opt-Out**

https://github.com/openshift/console/assets/102503482/fac30a1b-7c28-418f-a338-3cd0443ddb32

-------

**Cluster telemetry configuration description:** 

**ENFORCE**: Will work as the existing behaviour, will send telemetry events always. 
**DISABLED**: Will disable sending telemetry events. 
**OPT-OUT**: Will allow users to Accept or Deny sending telemetry events in user preference page or on the global notifier section added in the footer of the page with `Accept` and `Deny` buttons. Until user deny, telemetry events will be sent.
**OPT-IN**: Will allow users to Accept or Deny sending telemetry events in user preference page or on the global notifier section added in the footer of the page with `Accept` and `Deny` buttons and we will store last 10 events and send all to telemetry if user selects `Accept`

This value will be stored in window.SERVER_FLAGS.telemetry.STATE and in `operator.openshift.io/v1` under `metadata.annotations` as `telemetry.console.openshift.io/STATE: OPT-IN`

-------

User can change the setting anytime in `User Preferences--> General tab-->Telemetry`

-------

To test it , you can use the below command.

`./bin/bridge -plugins -telemetry 'STATE=OPT-IN'
`

### CODE REVIEW

**Comment by lokanandaprabhu** [comment]
/cc @jerolimov 

Hi @jerolimov ,

Can you please provide descriptions for 

---Cluster configuration page---
Opt-in
Opt-out
Enforce
Disabled
Telemetry cluster configuration help text

---User Preference page---
Accept
Deny
Telemetry User Preference help text

I will update the PR accordingly.

File: frontend/packages/console-shared/src/hooks/useTelemetry.ts [hunk 1]
```diff
@@ -29,10 +37,53 @@ export const updateServerFlagsFromTests = () => {
...
+    null,
+    true,
+  );
+
   const [extensions] = useResolvedExtensions<TelemetryListener>(isTelemetryListener);
 
   return React.useCallback<TelemetryEventListener>(
     (eventType, properties) => {
+      if (
+        !window.SERVER_FLAGS.telemetry?.STATE ||
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/hooks/useTelemetry.ts:52]: remove this condition, if it's not set, we expect that telemetry on (starting with 4.16 at least)

**Comment by jerolimov** [comment]
This looks almost good. As discussed please address this small changes and then we can merge this:

1. Can you please check why is `yarn i18n` isn't generating new translations? I saw some missing translation warnings when opening the cluster configuration.
2. Test cases should contain a "what should happen". Best practice is that the tests cases creates a sentence including the it. Like: `it('should do....`
3. Remove the minus from the OPT-IN and OPT-OUT enum. That allows you to use `ANALYTICS.OPTIN` instead of `ANALYTICS['OPT-IN']`
4. In `useTelemetry` extract the code that resends events from the telemetry callback into an effect that depends just on the user settings value.
5. I'm missing the telemetry identify event. We should check if it was send on the latest master branch.

**Comment by lokanandaprabhu** [comment]
Hi @jerolimov ,

Updated the PR based on review comments and for point 5 I am able to see Identity event.


https://github.com/openshift/console/assets/102503482/7739c77e-bd88-4764-8c7c-01859c2fea4d

**Review by jerolimov — APPROVED** [review]
Tested it again on a cluster bot and everything is working as expected. :+1:

/lgtm

**Comment by jerolimov** [comment]
Only tests was fixed.

/lgtm

File: frontend/packages/console-telemetry-plugin/integration-tests/support/commands/index.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Include the cypress customized commands related files
+import '@console/cypress-integration-tests/support/selectors';
+import '@console/cypress-integration-tests/support/a11y';
+import '@console/cypress-integration-tests/support/login';
+import '@console/cypress-integration-tests/support/project';
+import '@console/cypress-integration-tests/support';
+import '@console/dev-console/integration-tests/support/commands/app';
+import '@console/dev-console/integration-tests/support/commands/hooks';
```
Comment (sanketpathak) [inline frontend/packages/console-telemetry-plugin/integration-tests/support/commands/index.ts:8]: Remove this line

**Comment by sanketpathak** [comment]
Verified with admin and non-admin users. Tests run green
Screenshot 2024-03-20 at 1 02 25 PM
Screenshot 2024-03-20 at 1 09 10 PM
Screenshot 2024-03-20 at 1 13 02 PM


/label qe-approved

**Pull request merged**